src/Encamina.Enmarcha.SemanticKernel/Internals/SemanticKernelCosineStringSimilarityComparer.cs

```diff
@@ -0,0 +1,63 @@
+﻿using Encamina.Enmarcha.AI.Abstractions;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Embeddings;
+
+namespace Encamina.Enmarcha.SemanticKernel.Internals;
+
+#pragma warning disable SKEXP0001 // Type is for evaluation purposes only and is subject to change or removal in future updates. Suppress this diagnostic to proceed.
+
+/// <summary>
+/// An implementation of the <see cref="IStringSimilarityComparer"/> interface that uses cosine similarity algorithm, using semantic kernel to generate the embeddings.
+/// </summary>
+internal sealed class SemanticKernelCosineStringSimilarityComparer : IStringSimilarityComparer
+{
+    private readonly Kernel kernel;
+    private readonly ITextEmbeddingGenerationService textEmbeddingGenerationService;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="SemanticKernelCosineStringSimilarityComparer"/> class.
+    /// </summary>
+    /// <param name="kernel">A valid instance of a <see cref="Kernel"/>.</param>
+    public SemanticKernelCosineStringSimilarityComparer(Kernel kernel)
+    {
+        this.kernel = kernel;
+
+        textEmbeddingGenerationService = kernel.GetRequiredService<ITextEmbeddingGenerationService>();
+    }
+
+    /// <inheritdoc />
+    public async Task<double> CompareAsync(string first, string second, CancellationToken cancellationToken)
+    {
+        var embeddingsFirst = await textEmbeddingGenerationService.GenerateEmbeddingAsync(first, kernel, cancellationToken);
+        var embeddingsSecond = await textEmbeddingGenerationService.GenerateEmbeddingAsync(second, kernel, cancellationToken);
+
+        return CalculateCosineSimilarity(embeddingsFirst, embeddingsSecond);
+    }
+
+    private static double CalculateCosineSimilarity(ReadOnlyMemory<float> vec1, ReadOnlyMemory<float> vec2)
```

Comment:
Perhaps to calculate cosine similarity, we can utilize the Microsoft System.Numerics.Tensors library (https://www.nuget.org/packages/System.Numerics.Tensors), which provides a method for this purpose => `TensorPrimitives.CosineSimilarity`. 
Furthermore, we are already using it in another project within this solution precisely for this calculation, so we avoid doing the same thing in two different ways :)
https://github.com/Encamina/enmarcha/blob/3bfaf2614728f4aa0282828211626a85c217817d/src/Encamina.Enmarcha.AI/TextSplitters/SemanticTextSplitter.cs#L172
Great job!